identity-appsupport/src/commonMain/kotlin/com/android/identity/appsupport/ui/consent/ConsentModalBottomSheet.kt

```diff
@@ -0,0 +1,465 @@
+                ) {
+                    Text(
+                        text = if (relyingParty.trustPoint?.displayName != null) {
+                            stringResource(
+                                Res.string.consent_modal_bottom_sheet_share_and_stored_by_known_requester,
+                                relyingParty.trustPoint.displayName!!
+                            )
+                        } else if (relyingParty.websiteOrigin != null) {
+                            stringResource(
+                                Res.string.consent_modal_bottom_sheet_share_and_stored_by_known_requester,
+                                relyingParty.websiteOrigin
+                            )
+                        } else {
+                            stringResource(Res.string.consent_modal_bottom_sheet_share_and_stored_by_unknown_requester)
+                        },
+                        style = MaterialTheme.typography.bodyMedium,
+                        fontWeight = FontWeight.Bold
+                    )
+                }
+                DataElementGridView(storedFields, dontUseColumns)
+            }
+        }
+    }
+    Spacer(modifier = Modifier.height(2.dp))
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(shape = RoundedCornerShape(0.dp, 0.dp, 16.dp, 16.dp))
+            .background(MaterialTheme.colorScheme.surfaceContainerLowest),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Column(
+            modifier = Modifier.padding(16.dp)
+        ) {
+            // TODO: When we upgrade to a newer version of compose-ui we can use
+            //  AnnotatedString.fromHtml() and clicking the links will also work.
+            //  See b/139326648 for details.
+            //
+            val annotatedLinkString = buildAnnotatedString {
+                val str = stringResource(Res.string.consent_modal_bottom_sheet_wallet_privacy_policy)
+                val startIndex = 4
+                val endIndex = startIndex + 31
+                append(str)
+                addStyle(
+                    style = SpanStyle(
+                        color = Color(0xff64B5F6),
+                        textDecoration = TextDecoration.Underline
+                    ), start = startIndex, end = endIndex
+                )
+            }
+            Text(
+                text = annotatedLinkString,
+                style = MaterialTheme.typography.bodySmall
+            )
+        }
+    }
+    if (relyingParty.trustPoint == null) {
+        Box(
+            modifier = Modifier.padding(vertical = 8.dp)
+        ) {
+            WarningCard(
+                stringResource(Res.string.consent_modal_bottom_sheet_warning_verifier_not_in_trust_list)
+            )
+        }
+    }
+}
+
+@Composable
+private fun DataElementGridView(
+    consentFields: List<ConsentField>,
+    dontUseColumns: Boolean
+) {
+    // If we have less than 5 data elements, don't use columns.
```

Comment:
In case the number changes, leave it out of the comment, so it doesn't drift out of sync?
// If we have too few data elements....

Reply:
I just removed the entire comment.